src/Javascript/index.js

```diff
@@ -70,22 +70,38 @@ const app = Elm.Main.init({
 
 app.ports.copyToClipboard.subscribe(copyToClipboard)
 app.ports.deauthenticate.subscribe(deauthenticate)
+app.ports.fsDownloadItem.subscribe(fs.downloadItem)
 app.ports.showNotification.subscribe(showNotification)
 
 app.ports.redirectToLobby.subscribe(() => {
   wn.redirectToLobby(PERMISSIONS, location.origin + location.pathname)
 })
 
+
 exe("fsAddContent", "add")
 exe("fsListDirectory", "listDirectory")
 exe("fsListPublicDirectory", "listPublicDirectory")
 exe("fsMoveItem", "moveItem", { listParent: true })
 exe("fsRemoveItem", "removeItem", { listParent: true })
 
-app.ports.fsDownloadItem.subscribe(fs.downloadItem)
+
+registerServiceWorker({
+  path: "service-worker.js",
+  onUpdateAvailable: () => {
+    console.log("⚙️ Application update available")
+    app.ports.appUpdateAvailable.send(null)
+  },
+  onUpdateFinished: () => {
+    console.log("⚙️ Application update finished")
+    app.ports.appUpdateFinished.send(null)
+  },
+})
```

Comment:
The important part 👀

Reply:
Love it! :heart: 

I'm thinking about whether it makes sense to support something like showing the user "You're definitely on the latest version!". But I'd just wait until we've put this onto staging and we see how it feels. Maybe we don't even need that and `onUpdateAvailable` would trigger so quickly you would know about an update already if there was one.